src/Text/Layout/Table/Spec/RowGroup.hs

```diff
@@ -1,18 +1,61 @@
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE DeriveFoldable #-}
 module Text.Layout.Table.Spec.RowGroup where
 
 import Text.Layout.Table.Spec.Util
 
+import Data.List (transpose)
+import Data.Functor (void)
+
 -- | Groups rows together which should not be visually seperated from each other.
-newtype RowGroup a
-    = RowGroup
-    { rows :: [Row a]
-    }
+data RowGroup a
+    = MultiRowGroup [Row a]
+    | NullableRowGroup [Row (Maybe a)]
 
 -- | Group the given rows together.
 rowsG :: [Row a] -> RowGroup a
-rowsG = RowGroup
+rowsG = MultiRowGroup
 
 -- | Make a group of a single row.
 rowG :: Row a -> RowGroup a
-rowG = RowGroup . (: [])
+rowG = MultiRowGroup . (: [])
+
+-- | Provide a 'RowGroup' where single cells may be missing.
+optionalRowsG :: [Row (Maybe a)] -> RowGroup a
```

Comment:
Since you changed the constructor name, maybe this should be changed to match? `nullableRowsG`?